### PR TITLE
Threw error for post-processor/scalar output mismatch in CSV

### PR DESCRIPTION
**Pull request by joshuahansel**
Refs #25211

### CODE REVIEW

**Comment by joshuahansel** [comment]
@loganharbour You're an output guy, right? Mind reviewing?

File: framework/src/outputs/CSV.C [hunk 1]
```diff
@@ -90,6 +90,21 @@ CSV::initialSetup()
       MooseUtils::clearSymlink(out_final);
     }
   }
+
+  // See https://github.com/idaholab/moose/issues/25211.
+  const auto pp_execute_on = advancedExecuteOn().find("postprocessors")->second;
+  const auto scalar_execute_on = advancedExecuteOn().find("scalars")->second;
+  const auto reporter_execute_on = advancedExecuteOn().find("reporters")->second;
```
Comment (loganharbour) [inline framework/src/outputs/CSV.C:97]: Use `libmesh_map_find` for these, so we at least die horribly if we can't find these

File: test/tests/outputs/csv/tests [hunk 2]
```diff
@@ -105,4 +105,12 @@
...
     issues = "#24705"
   []
+  [pp_scalar_execute_on_mismatch]
+    type = RunException
+    input = 'csv.i'
+    cli_args = 'Outputs/csv/execute_postprocessors_on=FINAL'
+    expect_err = "The parameters 'execute_postprocessors_on', 'execute_scalars_on', and 'execute_reporters_on' must be the same for CSV output"
+    requirement = "The system shall report an error if post-processors and scalar variables have different output frequency."
+    issues = "#25211"
+  []
```
Comment (loganharbour) [inline test/tests/outputs/csv/tests:115]: I know it's not great.. but aren't there multiple combos to test here?

**Review by loganharbour — CHANGES_REQUESTED** [review]
Not great, not sure how I'd make it better. So - I think this works

File: test/tests/outputs/csv/tests (hunk 2 above)
Comment (joshuahansel) [reply test/tests/outputs/csv/tests:115]: Yes, I'd like to choose only the most important combos. Someone may address the issue in a more elegant manner (probably not any time soon though...). What tests would you like?

Comment (loganharbour) [reply test/tests/outputs/csv/tests:115]: Let's just add one for reporters and that's good enough

File: framework/src/outputs/CSV.C (hunk 1 above)
Comment (joshuahansel) [reply framework/src/outputs/CSV.C:97]: Am I doing something wrong?
```
/Users/hansje/projects/sockeye/moose/framework/src/outputs/CSV.C:95:30: error: no matching function for call to 'map_find'
  const auto pp_execute_on = libmesh_map_find(advancedExecuteOn(), "postprocessors");
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:39:36: note: expanded from macro 'libmesh_map_find'
#define libmesh_map_find(map, key) libMesh::Utility::map_find((map), (key), __FILE__, __LINE__)
                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:89:1: note: candidate template ignored: requirement '!is_streamable<char [15]>::value' was not satisfied [with Map = const OutputOnWarehouse, Key = char [15]]
map_find(Map & map,
^
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:108:1: note: candidate template ignored: requirement '!is_streamable<char [15]>::value' was not satisfied [with Map = OutputOnWarehouse, Key = char [15]]
map_find(const Map & map,
^
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:128:1: note: candidate template ignored: substitution failure [with Map = const OutputOnWarehouse, Key = char [15], $2 = nullptr]: no type named 'mapped_type' in 'OutputOnWarehouse'
map_find(Map & map,
^
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:147:1: note: candidate template ignored: substitution failure [with Map = OutputOnWarehouse, Key = char [15], $2 = nullptr]: no type named 'mapped_type' in 'OutputOnWarehouse'
map_find(const Map & map,
```
Does `libmesh_map_find` work with this wrapped map class `OutputMapWrapper`?

Comment (loganharbour) [reply framework/src/outputs/CSV.C:97]: Oops. It doesn't. Just add the following above then:

```
mooseAssert(advancedExecuteOn().contains("postprocessors"), "Missing expected postprocessors key");
mooseAssert(advancedExecuteOn().contains("scalars"), "Missing expected scalars key");
mooseAssert(advancedExecuteOn().contains("reporters"), "Missing expected reporters key");
```

**Review by loganharbour — APPROVED** [review]

**Comment by joshuahansel** [comment]
@loganharbour Do I need to worry about the libtorch target that keeps failing, or am I good to merge?

**Pull request merged**